### DISCUSSION TITLE
How Are Rebranded Instruments Handled?

### DISCUSSION THREAD

**Discussion by fundthmcalculus** [Galvant/InstrumentKit#391] [Q&A] [answered]
The primary question in my use case is the Hewlett-Packard -> Agilent -> VTI -> Ametek -> Keysight (or some permutation thereof) sequence. I'm about to start adding a few drivers, and wanted to know under what name to locate them.
1. Use the oldest name under which the instrument existed? (eg, HP E3643A)
2. Use the name under which *my* instrument exists? (Agilent E3643A) 
3. Use the current name under which the instrument exists (if any)? (Keysight E3643A)

My inclination is to go with 1 or 3, but I couldn't find specific information one way or the other. Or does `instrumentkit` have a specific policy/recommendation on this question?

**Comment by trappitsch** [comment]
A fantastic question that has come up a long time ago already and seems unsolved at this point, see #35. Great time to pick it back up, what's your opinion @scasagrande. You had some time now to think about it :zany_face:

**Reply by fundthmcalculus** [reply]
I read the other discussion thread, and I tend to think that we should add wrappers for the instruments under any name in which they existed. I know it makes the name spacing more complex, but it will maximize the exposure for anyone who is not familiar with the history.

Another option might be to put it under one namespace, and make a Read me file in the other ones that say if you are looking for this instrument it's located under [HP/etc].

**Reply by scasagrande** [reply, marked as answer]
Sorry for the delay, life etc etc

We can revive the discussion in the linked issue, but in general I lean towards allowing an instrument to resolve under any branding by which it was available from the manufacturer. Having a user know that their particular instrument is the same as some other instrument, with no clear way to know which namespace to look under isn't a good user experience. 

From a developer's perspective, there is also the issue: "under which namespace does the implementation live?". I lean towards "the oldest by which that instrument was available". Then the re-badging can be handled as either slim wrappers in their own python files, or we can come up with some sort of metadata to expand these namespaces at runtime. The former is much easier and quicker to implement, and can also be clearer to someone less familiar with the source code where to look in order to make changes.

**Reply by fundthmcalculus** [reply]
@scasagrande no worries! I totally agree with all your points. I was conflating the two points of [library user experience] and [library maintainer experience]. I'll update the linked PR with those points in mind. For other HP/Agilent/VTI/Ametek instruments where I am aware of handover, I'll make a separate PR.

**Reply by scasagrande** [reply]
Awesome thanks @fundthmcalculus !!
